api/v1beta2/awsmachine_webhook.go

```diff
@@ -113,6 +113,20 @@ func (r *AWSMachine) ValidateUpdate(old runtime.Object) error {
 		delete(cloudInit, "secureSecretsBackend")
 	}
 
+	// allow limited changes to instanceDetails if we are defaulting and the field was missing
+	// because we know what the field should contain
+	_, oldInstanceDetails := oldAWSMachineSpec["instanceDetails"]
+	_, newInstanceDetails := newAWSMachineSpec["instanceDetails"]
+	if oldInstanceDetails != newInstanceDetails {
+		instanceDetails := map[string]interface{}{
+			"instanceType": oldAWSMachineSpec["instanceType"],
+		}
+		if _, ok := oldAWSMachineSpec["spotMarketOptions"]; ok {
+			instanceDetails["spotMarketOptions"] = oldAWSMachineSpec["spotMarketOptions"]
+		}
+		oldAWSMachineSpec["instanceDetails"] = []interface{}{instanceDetails}
+	}
+
 	if !cmp.Equal(oldAWSMachineSpec, newAWSMachineSpec) {
```

Comment:
Errr, correct me if I'm wrong, but if the previous thing changes something in the machine spec because of the re-order, shouldn't this thing fail here?